include/zephyr/sys/kobject.h

```diff
@@ -192,6 +192,17 @@ __syscall void k_object_release(const void *object);
  */
 void k_object_access_all_grant(const void *object);
 
+/**
+ * Check if a kernel object is of certain type and is valid.
+ *
+ * This checks if the kernel object exists, of certain type,
+ * and has been initialized.
+ *
+ * @param obj Address of the kernel object
+ * @param otype Object type (use K_OBJ_ANY for all types)
+ */
```

Comment:
Missing documentation for the `@retval`s of this function

Reply:
Use `@return` instead as I cannot think of an easy way to convey `@retval false`.